app/hooks/useTrucoGame.ts

```diff
@@ -218,6 +243,98 @@ export function useTrucoGame() {
     }
   }, [gameState]);
 
+  const handleEnvido = useCallback(async () => {
+    // Can only offer envido in the first turn of a round and before truco
+    if (gameState.phase.type !== 'HUMAN_TURN' || 
+        gameState.playedCards.length > 0 || 
+        gameState.trucoState.type !== 'NONE' ||
+        gameState.envidoState.type !== 'NONE') {
+      console.error('Error: Attempted to offer envido in invalid game state. This should not be possible.');
+      return;
+    }
+    
+    // Cannot offer envido when someone has flor - use original cards for accurate flor detection
+    const humanHasFlor = hasFlor(gameState.originalHumanCards, gameState.muestraCard);
+    const aiHasFlor = hasFlor(gameState.originalAiCards, gameState.muestraCard);
+    
+    if (humanHasFlor || aiHasFlor) {
+      console.error('Error: Attempted to offer envido when there is flor. This should not be possible.');
+      setGameState(prev => ({
+        ...prev,
+        message: 'No se puede cantar envido cuando hay flor.'
+      }));
+      return;
+    }
+
+    
+    // Calculate envido points
+    const humanEnvidoPoints = calculateEnvidoPoints(gameState.humanCards, gameState.muestraCard);
+    const aiEnvidoPoints = calculateEnvidoPoints(gameState.aiCards, gameState.muestraCard);
+    
+    setGameState(prev => ({
+      ...prev,
+      envidoState: { 
+        type: 'CALLED', 
+        lastCaller: 'HUMAN',
+        humanPoints: humanEnvidoPoints,
+        aiPoints: aiEnvidoPoints 
+      },
+      message: 'Gritaste envido! El jugador CPU está decidiendo...'
+    }));
+
+    const aiDecision = await getEnvidoOfferAIDecision(
+      gameState.aiCards,
+      gameState.humanCards,
```

Comment:
I'm sure we want to use the original cards in the players' hand (e.g. gameState.originalAiCards)